opentelemetry-api/src/opentelemetry/metrics/__init__.py

```diff
@@ -178,60 +75,150 @@ def get_handle(self, label_values: List[str]) -> "object":
+    should be used when the metric cannot be expressed as a sum or because
+    the measurement interval is arbitrary.
+    """
 
-    def get_handle(self, label_values: List[str]) -> "CounterHandle":
-        """Gets a `CounterHandle` with an int value."""
+    def get_handle(self, label_values: Sequence[str]) -> "GaugeHandle":
+        """Gets a `GaugeHandle`."""
+        return GaugeHandle()
 
 
-class FloatGauge(Metric):
-    """A gauge type metric that holds float values."""
+class Measure(Metric):
+    """A measure type metric that represent raw stats that are recorded.
 
-    def get_handle(self, label_values: List[str]) -> "GaugeHandle":
-        """Gets a `GaugeHandle` with a float value."""
+    Measure metrics represent raw statistics that are recorded. By
+    default, measure metrics can accept both positive and negatives.
+    Negative inputs will be discarded when monotonic is True.
+    """
 
+    def get_handle(self, label_values: Sequence[str]) -> "MeasureHandle":
+        """Gets a `MeasureHandle` with a float value."""
+        return MeasureHandle()
 
-class IntGauge(Metric):
-    """A gauge type metric that holds int values."""
 
-    def get_handle(self, label_values: List[str]) -> "GaugeHandle":
-        """Gets a `GaugeHandle` with an int value."""
+MetricT = TypeVar("MetricT", Counter, Gauge, Measure)
 
 
-class FloatMeasure(Metric):
-    """A measure type metric that holds float values."""
+# pylint: disable=unused-argument
+class Meter:
+    """An interface to allow the recording of metrics.
 
-    def get_handle(self, label_values: List[str]) -> "MeasureHandle":
-        """Gets a `MeasureHandle` with a float value."""
+    `Metric` s are used for recording pre-defined aggregation (gauge and
+    counter), or raw values (measure) in which the aggregation and labels
+    for the exported metric are deferred.
+    """
+
+    def record_batch(
+        self,
+        label_values: Sequence[str],
+        record_tuples: Sequence[Tuple["Metric", ValueT]],
+    ) -> None:
+        """Atomically records a batch of `Metric` and value pairs.
 
+        Allows the functionality of acting upon multiple metrics with
+        a single API call. Implementations should find metric and handles that
+        match the key-value pairs in the label tuples.
+
+        Args:
+            label_values: The values that will be matched against to record for
+                the handles under each metric that has those labels.
+            record_tuples: A sequence of pairs of `Metric` s and the
+                corresponding value to record for that metric.
+        """
+
+    def create_metric(
+        self,
+        name: str,
+        description: str,
+        unit: str,
+        value_type: Type[ValueT],
+        metric_type: Type[MetricT],
+        label_keys: Sequence[str] = None,
+        enabled: bool = True,
+        monotonic: bool = False,
+    ) -> "Metric":
```

Comment:
```suggestion
    ) -> MetricT:
```

(see comment in sdk's create_metric)